NYTPhotoViewer/NYTInterstitialViewController.m

```diff
@@ -37,17 +37,27 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder {
     return self;
 }
 
+- (void)prepareLayout {
+    if (self.interstitialView.translatesAutoresizingMaskIntoConstraints) {
+        self.interstitialView.center = CGPointMake(CGRectGetMidX(self.view.bounds), CGRectGetMidY(self.view.bounds));
+    } else {
+        [NSLayoutConstraint activateConstraints:@[
+            [self.interstitialView.centerXAnchor constraintEqualToAnchor:self.view.centerXAnchor],
+            [self.interstitialView.centerYAnchor constraintEqualToAnchor:self.view.centerYAnchor],
+        ]];
+    }
+}
+
 - (void)viewDidLoad {
     [super viewDidLoad];
 
     [self.view addSubview:self.interstitialView];
-    self.interstitialView.center = CGPointMake(CGRectGetMidX(self.view.bounds), CGRectGetMidY(self.view.bounds));
+    [self prepareLayout];
 }
 
 - (void)viewWillLayoutSubviews {
     [super viewWillLayoutSubviews];
-
-    self.interstitialView.center = CGPointMake(CGRectGetMidX(self.view.bounds), CGRectGetMidY(self.view.bounds));
+    [self prepareLayout];
```

Comment:
This will recreate the constraints on ever `viewWillLayoutSubviews`. Probably best to hold them in an array property and only construct them if they are nil.